packages/graphback-core/src/scalars/index.ts

```diff
@@ -1,2 +1,70 @@
-export * from 'graphql-type-json';
-export * from './scalars'
+import { GraphQLNamedType, GraphQLScalarType } from 'graphql';
+import { TimeResolver, TimestampResolver, DateResolver, DateTimeResolver, ObjectIDResolver, JSONResolver, JSONObjectResolver } from 'graphql-scalars';
+
+export const GraphbackTime = new GraphQLScalarType({
+  ...extractConfig(TimeResolver),
+  name: "GraphbackTime"
+});
+
+export const GraphbackTimestamp = new GraphQLScalarType({
+  ...extractConfig(TimestampResolver),
+  name: "GraphbackTimestamp"
+});
+
+export const GraphbackDate = new GraphQLScalarType({
+  ...extractConfig(DateResolver),
+  name: "GraphbackDate"
+});
+
+export const GraphbackDateTime = new GraphQLScalarType({
+  ...extractConfig(DateTimeResolver),
+  name: "GraphbackDateTime"
+});
+
+// TODO - modify parseValue and parseLiteral so that we deserialise into Mongo ObjectID instead of a string.
+export const GraphbackObjectID = new GraphQLScalarType({
+  ...extractConfig(ObjectIDResolver),
+  name: "GraphbackObjectID"
+});
```

Comment:
I feel like `parseLiteral `and `parseValue` should directly parse the string to [ObjectID](https://docs.mongodb.com/manual/reference/method/ObjectId/) type. What graphql-scalars does is only performing the validations against the argument and returning the string. For us going on a step further would mean the the MongoDB Data Provider would not have to do stuff like:
-   https://github.com/aerogear/graphback/blob/2b77fb445ad245eec00f92999b77b126586871b4/packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts#L66
- https://github.com/aerogear/graphback/blob/2b77fb445ad245eec00f92999b77b126586871b4/packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts#L101
- Or modifying the [query builder](https://github.com/aerogear/graphback/blob/2b77fb445ad245eec00f92999b77b126586871b4/packages/graphback-runtime-mongodb/src/queryBuilder.ts) to make ObjectID filters work, something observed when trying the things described in https://github.com/aerogear/graphback/issues/1625#issuecomment-660083625

Reply:
What do you think? @craicoverflow @wtrocki

Reply:
+100 from me but I feel that GraphQL-Scalars (Urigo) Might want that as well. maybe worth creating issue on their side

Reply:
> +100 from me but I feel that GraphQL-Scalars (Urigo) Might want that as well. maybe worth creating issue on their side.
 
> There is easy way to introduce this without breaking GraohQL Scalars. We do not want to parse object id twice - for validation and actual object

In order to be able to parse to actual `ObjectID` we'll need to use the [mongodb](https://www.npmjs.com/package/mongodb) package. Not sure, how to approach this: peerDependency?

Reply:
Yes.. That will work

Reply:
I have opened https://github.com/Urigo/graphql-scalars/issues/429 upstream. While waiting for a response, I'd do it on our side.

Reply:
Awesome!